confgenerator/files.go

```diff
@@ -61,3 +61,42 @@ func GenerateFiles(input, service, logsDir, stateDir, outDir string) error {
 	}
 	return nil
 }
+
+// TODO(davidbtucker): Refacator the functions in this file.  We should have a single unified config (not per-platform).
+func GenerateFilesFromUnifiedConfigWindows(unified *UnifiedConfigWindows, service, logsDir, stateDir, outDir string) error {
```

Comment:
Is this part of the code somehow related to https://github.com/GoogleCloudPlatform/ops-agent/blob/01bcc3eb52e74eadca105313c29541b20e3c6d5a/confgenerator/files.go#L10? Seems like similar logic is there as well. Any reason we need to modify config generation when detecting service conflicts? Or is this just some refactoring?

Reply:
Right, this is just some refactoring.  At a high level, we want:

1. Read the Ops Agent config
2. Do any checks on the Ops Agent config (e.g. whether it conflicts with standalone agents)
3. From the Ops Agent config, generate the sub-agent configs.

The existing function does all steps 1-3.  The new function just does step 3.

We should refactor these functions to eliminate the duplicate logic, but that will be easier once we remove the per-platform Ops Agent config definitions.

Reply:
Got it. Can we create a bug for this with this context and add a TODO at https://github.com/GoogleCloudPlatform/ops-agent/blob/01bcc3eb52e74eadca105313c29541b20e3c6d5a/confgenerator/files.go#L10 as well?

Reply:
Yep good idea, done.

Reply:
The TODO is fixed in https://github.com/GoogleCloudPlatform/ops-agent/pull/42

Reply:
Thanks Quentin! That simplified this PR a lot.